packages/core/src/utils/ProfileUtils.ts

```diff
@@ -0,0 +1,107 @@
+ * created using Zowe CLI, and not rely on base profiles
+ *
+ * @param {string} profileType - The name of the profile type
+ * @param {boolean} mergeWithBase - Whether or not to merge with the base profile
+ * @returns {IProfile} - The default profile (or merged profile, if requested)
+ */
+export async function getDefaultProfile(profileType: string, mergeWithBase?: boolean) {
+
+    const profileRootDir: string = path.join(getZoweDir(), "profiles");
+    const logger: Logger = Logger.getImperativeLogger();
+    let profileManager: CliProfileManager;
+    let profileLoaded: IProfileLoaded;
+
+    try {
+        // Create the profile manager
+        profileManager = new CliProfileManager({
+            profileRootDirectory: profileRootDir,
+            type: profileType
+        });
+
+        // Load the profile using the profile manager
+        profileLoaded = await profileManager.load({
+            loadDefault: true
+        })
+    } catch (err) {
+        logger.warn(err.message);
+    }
+
+    // Throw an error if there was no default profile found and no base profile requested
+    if ((!profileLoaded || !profileLoaded.profile) && !mergeWithBase) {
+        throw new ImperativeError({msg: `Failed to load default profile of type "${profileType}"`});
+    }
+
+    // Give the profile back as-is if the profile is not to be merged with the base profile
+    if (mergeWithBase === false) {
+        return profileLoaded.profile;
+    }
+
+    let baseProfileManager: CliProfileManager;
+    let baseProfileLoaded: IProfileLoaded;
+
+    try {
+        baseProfileManager = new CliProfileManager({
+            profileRootDirectory: profileRootDir,
+            type: "base"
+        });
+
+        baseProfileLoaded = await baseProfileManager.load({
+            loadDefault: true
+        });
+    } catch (err) {
+        logger.warn(err.message);
+    }
+
+    // Return service profile if there was no base profile found and service profile existed.
+    // Return base profile if there was no service profile found and base profile existed.
+    // If both exist, combine. Otherwise, error
+    if ((!baseProfileLoaded || !baseProfileLoaded.profile) && (profileLoaded && profileLoaded.profile)) {
+        return profileLoaded.profile;
+    } else if (baseProfileLoaded && baseProfileLoaded.profile && (!profileLoaded || !profileLoaded.profile)) {
+        return baseProfileLoaded.profile;
+    } else if ((!baseProfileLoaded || !baseProfileLoaded.profile) && (!profileLoaded || !profileLoaded.profile)) {
+        throw new ImperativeError({msg: `Failed to load default profiles of types "${profileType}" and "base"`});
+    }
+
+    const combinedProfile: IProfile = JSON.parse(JSON.stringify(baseProfileLoaded.profile));
+
+    for (const propertyName in profileLoaded.profile) {
+        // tslint:disable-next-line: triple-equals
+        if (profileLoaded.profile[propertyName] != undefined) {
```

Comment:
If the condition is changed to `profileLoaded.profile[propertyName] != null`, it should behave the same and the TSLint comment above should be unnecessary. Not a big deal though 🙂